src/NServiceBus.Core.Tests/Msmq/MsmqQueueCreatorTests.cs

```diff
@@ -0,0 +1,256 @@
+﻿namespace NServiceBus.Core.Tests.Msmq
+{
+    using System;
+    using System.Messaging;
+    using System.Security.Principal;
+    using NUnit.Framework;
+    using Transport;
+
+    [TestFixture]
+    public class MsmqQueueCreatorTests
+    {
+        [Test]
+        public void Should_create_all_queues()
+        {
+            var creator = new QueueCreator(true);
+            var bindings = new QueueBindings();
+
+            DeleteQueueIfPresent("MsmqQueueCreatorTests.receiver");
+            DeleteQueueIfPresent("MsmqQueueCreatorTests.endpoint1");
+            DeleteQueueIfPresent("MsmqQueueCreatorTests.endpoint2");
+
+            bindings.BindReceiving("MsmqQueueCreatorTests.receiver");
+            bindings.BindSending("MsmqQueueCreatorTests.target1");
+            bindings.BindSending("MsmqQueueCreatorTests.target2");
+
+            creator.CreateQueueIfNecessary(bindings, WindowsIdentity.GetCurrent().Name);
+
+            Assert.True(QueueExists("MsmqQueueCreatorTests.receiver"));
+            Assert.True(QueueExists("MsmqQueueCreatorTests.target1"));
+            Assert.True(QueueExists("MsmqQueueCreatorTests.target2"));
+        }
+
+        [Test]
+        public void Should_not_create_remote_queues()
+        {
+            var creator = new QueueCreator(true);
+            var bindings = new QueueBindings();
+
+            bindings.BindSending("MsmqQueueCreatorTests.target3@remote");
+
+            creator.CreateQueueIfNecessary(bindings, WindowsIdentity.GetCurrent().Name);
+
+            Assert.False(QueueExists("MsmqQueueCreatorTests.target3"));
```

Comment:
Feels strange to pass in `MsmqQueueCreatorTests.target3@remote` and then assert `MsmqQueueCreatorTests.target`